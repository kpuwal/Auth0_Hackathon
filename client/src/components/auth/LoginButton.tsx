import { useAuth0 } from "@auth0/auth0-react";

interface IloginButton {
  signup?: boolean;

}

const LoginButton = ({signup}: IloginButton) => {
  const { loginWithRedirect } = useAuth0();

  return (
    <>
    {
      signup ?
      <button 
        className="nav-button" 
        id="activated" 
        onClick={() => loginWithRedirect({
          screen_hint: "signup",
        })}
      >
        Sign Up
      </button>
      :
      <button 
        className="nav-button"
        id="activated"
        onClick={() => loginWithRedirect()}
      >
        Log In
      </button>
    }
    {/* // <button className="nav-button" id="activated" onClick={() => loginWithRedirect()}>Log In</button> */}
    </>
  )
};

export default LoginButton;
