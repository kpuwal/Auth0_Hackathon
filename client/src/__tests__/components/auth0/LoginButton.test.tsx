import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
// import App from '../App';
// import Auth0ProviderWithHistory from '../components/auth/Auth0ProviderWithHistory';
import LoginButton from  '../../../components/auth/LoginButton';

// jest.mock("@auth0/auth0-react", () => ({
//     withAuthenticationRequired: jest.fn().mockImplementation((component, ignore) => component)
// }))

// test('renders profile view', () => {
//     const rendered = render(<Profile/>);
//     const sectionHeading = rendered.container.querySelector("#profile-section-heading")
//     expect(sectionHeading).toBeInTheDocument()
// });

// afterEach(() => jest.clearAllMocks())

describe('Login Button', () => {
  it ('renders login button', () => {
    const component = shallow(<LoginButton />);
    expect(component).toMatchSnapshot();
  });

  it('simulates click events', () => {
    const onButtonClick = sinon.spy();
    const wrapper = mount((
      <button onClick={onButtonClick} />
    ));
    wrapper.find('button').simulate('click');
    expect(onButtonClick).toHaveProperty('callCount', 1);
  });
});
