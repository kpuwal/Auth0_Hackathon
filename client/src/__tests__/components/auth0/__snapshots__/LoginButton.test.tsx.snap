// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Button renders login button 1`] = `
<Fragment>
  <button
    className="nav-button"
    id="inactive"
    onClick={[Function]}
  >
    Log In
  </button>
</Fragment>
`;
